oauth2_http/java/com/google/auth/oauth2/LoggingUtils.java

```
@@ -0,0 +1,211 @@
+                    headers.put(key, hashedVal);
+                  } else {
+                    headers.put(key, val);
+                  }
+                });
+        loggingDataMap.put("request.headers", headers.toString());
+
+        if (request.getContent() != null && logger.isDebugEnabled()) {
+          // are payload always GenericData? If so, can parse and store in json
+          if (request.getContent() instanceof UrlEncodedContent) {
+            // this is parsed to GenericData because that is how it is constructed.
+            GenericData data = (GenericData) ((UrlEncodedContent) request.getContent()).getData();
+            Map<String, String> contextMap = parseGenericData(data);
+            loggingDataMap.put("request.payload", contextMap.toString());
+          } else if (request.getContent() instanceof JsonHttpContent) {
+            String data = ((JsonHttpContent) request.getContent()).getData().toString();
+            loggingDataMap.put("request.payload", data);
+          }
+
+          logWithMDC(logger, org.slf4j.event.Level.DEBUG, loggingDataMap, message);
+        } else {
+
+          logWithMDC(logger, org.slf4j.event.Level.INFO, loggingDataMap, message);
+        }
+      }
+    } catch (Exception e) {
+      logger.error("Error logging request: ", e);
+    }
+  }
+
+  static void logResponse(HttpResponse response, Logger logger, String message) {
+    try {
+      if (logger.isInfoEnabled()) {
+        Map<String, String> responseLogDataMap = new HashMap<>();
+        responseLogDataMap.put("response.status", String.valueOf(response.getStatusCode()));
+        responseLogDataMap.put("response.status.message", response.getStatusMessage());
+
+        Map<String, Object> headers = new HashMap<>(response.getHeaders());
+        responseLogDataMap.put("response.headers", headers.toString());
+        logWithMDC(logger, org.slf4j.event.Level.INFO, responseLogDataMap, message);
+      }
+    } catch (Exception e) {
```

Similar to the [comment in sdk-platform-java](https://github.com/googleapis/sdk-platform-java/pull/3403/files#r1881114202), I think we should try to avoid doing these try catches or fail more gracefully.

---

as discussed offline, we decided to stick with try catches to let logging fail silently for any logging exceptions.